### PR TITLE
[Intel GPU][Windows] Fix overriding default CMAKE_CXX_FLAGS

### PR DESCRIPTION
**Pull request by min-jean-cho**
The root cause is that `/EHsc` is part of the default `CMAKE_CXX_FLAGS` in CMake. 
Fix to not override the default `CMAKE_CXX_FLAGS`. 

cc @gujinghui @PenghuiCheng @XiaobingSuper @jianyuh @jgong5 @mingfeima @sanchitintel @ashokei @jingxu10 @yanbing-j @Guobing-Chen @Xia-Weiwen @snadampal

### CODE REVIEW

**Review by EikanWang — APPROVED** [review]

**Comment by EikanWang** [comment]
This PR intends to fix a potential issue for oneDNN Windows for Intel GPU by passing `/EHsc` explicitly.

We build oneDNN for Intel GPU on Windows by Intel SYCL compiler. But the default value of the SYCL compiler to handle C++ standard exception is `OFF`. It means the default exception unwinding code doesn't destroy automatic C++ objects outside of blocks that go out of scope because of an exception. Resource leaks and undefined behavior may result when a C++ exception is thrown.

**Comment by EikanWang** [comment]
@min-jean-cho , please update the PR description.

**Comment by min-jean-cho** [comment]
Thanks @EikanWang , updated PR description.

File: cmake/Modules/FindMKLDNN.cmake [hunk 1]
```diff
@@ -43,15 +43,15 @@ IF(NOT MKLDNN_FOUND)
       endif()
     endif()
     if(LINUX)
-      set(ABI_NEUTRAL_FLAGS -fpreview-breaking-changes)
+      set(DNNL_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fpreview-breaking-changes")
```
Comment (EikanWang) [inline cmake/Modules/FindMKLDNN.cmake:46]: @min-jean-cho , why did you remove the `else` branch? Current block is Linux only!

Comment (min-jean-cho) [reply cmake/Modules/FindMKLDNN.cmake]: @EikanWang , thanks for the catch. Fixed the logic.

File: cmake/Modules/FindMKLDNN.cmake [hunk 2]
```diff
@@ -42,16 +42,17 @@ IF(NOT MKLDNN_FOUND)
         list(APPEND DNNL_MAKE_COMMAND "--" "-l" "$ENV{MAX_JOBS}")
       endif()
     endif()
+    set(DNNL_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
```
Comment (fengyuan14) [inline cmake/Modules/FindMKLDNN.cmake:45]: I wonder if `CMAKE_CXX_FLAGS` here still is the default one. In PyTorch cmake system, there are a lot of adds in `CMAKE_CXX_FLAGS`. If `CMAKE_CXX_FLAGS` is not the default one here, is it risky involve PyTorch added flags in oneDNN build?

Comment (min-jean-cho) [reply cmake/Modules/FindMKLDNN.cmake]: @fengyuan14 , thanks. It's a valid point. Updated.

**Review by EikanWang — APPROVED** [review]

**Comment by EikanWang** [comment]
Hi @atalman , this failure is irrelevant to this PR. we also found the issue on PyTorch main branch for both CUDA and MacOS.

https://github.com/pytorch/pytorch/actions/runs/10706859375/job/29696277867
https://github.com/pytorch/pytorch/actions/runs/10706859375/job/29696276406

**Review by atalman — APPROVED** [review]
